datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/util/OwnerUtils.java

```diff
@@ -150,76 +136,79 @@ private static void addOwner(
             : com.linkedin.common.OwnershipType.CUSTOM;
 
     newOwner.setType(gmsType);
-    newOwner.setTypeUrn(ownershipUrn);
+    newOwner.setTypeUrn(ownershipTypeUrn);
     newOwner.setSource(new OwnershipSource().setType(OwnershipSourceType.MANUAL));
     newOwner.setOwner(ownerUrn);
     ownerArray.add(newOwner);
     ownershipAspect.setOwners(ownerArray);
   }
 
+  private static void removeExistingOwnerIfExists(
+      OwnerArray ownerArray, Urn ownerUrn, Urn ownershipTypeUrn) {
+    ownerArray.removeIf(
+        owner -> {
+          // Remove old ownership if it exists (check ownerUrn + type (entity & deprecated type))
+          return isEqual(owner, ownerUrn, ownershipTypeUrn);
+        });
+  }
+
+  public static boolean isEqual(Owner owner, Urn ownerUrn, Urn ownershipTypeUrn) {
```

Comment:
it would be nice to have some unit tests for this utils file as well! a good amount of basic logic we could be testing here